dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/LuceneQueryDateTimeFormatter.java

```diff
@@ -0,0 +1,253 @@
+                        .toLowerCase();
+
+                if (clause.startsWith(structureVarName + "." + field.getVelocityVarName().toLowerCase() + ":") || clause.startsWith("moddate:")) {
+                    replace = new String(clause);
+                    if (field.getFieldType().equals(Field.FieldType.DATE_TIME.toString()) || clause.startsWith("moddate:")) {
+                        matches = RegEX.find(replace, "\\[(\\d{1,2}/\\d{1,2}/\\d{4}) TO ");
+                        for (RegExMatch regExMatch : matches) {
+                            replace = replace.replace("[" + regExMatch.getGroups().get(0).getMatch() + " TO ", "["
+                                    + regExMatch.getGroups().get(0).getMatch() + " 00:00:00 TO ");
+                        }
+
+                        matches = RegEX.find(replace, " TO (\\d{1,2}/\\d{1,2}/\\d{4})\\]");
+                        for (RegExMatch regExMatch : matches) {
+                            replace = replace.replace(" TO " + regExMatch.getGroups().get(0).getMatch() + "]", " TO "
+                                    + regExMatch.getGroups().get(0).getMatch() + " 23:59:59]");
+                        }
+                    }
+
+                    //if structureVarName or field.getVelocityVarName() has numbers, we need to split the clause to format only dates
+                    if (clause.startsWith(
+                            structureVarName + "." + field.getVelocityVarName().toLowerCase()
+                                    + ":")) {
+                        replace = structureVarName + "." + field.getVelocityVarName()
+                                .toLowerCase() + ":" + replaceDateTimeFormatInClause(
+                                replace.substring(replace.indexOf(":") + 1));
+                    } else {
+                        replace = replaceDateTimeFormatInClause(replace);
+                    }
+
+                    query = query.replace(clause, replace);
+
+                    break;
+                }
+            }
+        }
+
+        matches = RegEX.find(query, "\\[([0-9]*)(\\*+) TO ");
+        for (RegExMatch regExMatch : matches) {
+            query = query.replace("[" + regExMatch.getGroups().get(0).getMatch() + regExMatch.getGroups().get(1).getMatch() + " TO ", "["
+                    + regExMatch.getGroups().get(0).getMatch() + " TO ");
+        }
+
+        matches = RegEX.find(query, " TO ([0-9]*)(\\*+)\\]");
+        for (RegExMatch regExMatch : matches) {
+            query = query.replace(" TO " + regExMatch.getGroups().get(0).getMatch() + regExMatch.getGroups().get(1).getMatch() + "]", " TO "
+                    + regExMatch.getGroups().get(0).getMatch() + "]");
+        }
+
+        matches = RegEX.find(query, "\\[([0-9]*) (TO) ([0-9]*)\\]");
+        if(matches.isEmpty()){
+            matches = RegEX.find(query, "\\[([a-z0-9]*) (TO) ([a-z0-9]*)\\]");
+        }
+        for (RegExMatch regExMatch : matches) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'regExMatch' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4545506)